### PR TITLE
Sentiment-user-app

### PR DESCRIPTION
**Pull request by ALeemans**
created a separate app that runs on its on. The app reads from csv and does a sentiment analysis on a predefined column. It returns a csv file with a sentiment score from -1 to 1 (negative to positive).

### CODE REVIEW

File: user-app/scripts/readfiles.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+def readCsv(csvfile,sep=';',nrows=None):
+    import os
+    from os import path
+    #check if file exists with csv extension in research drive of user
+    userRD = os.getlogin()
+    location = r'G:\My Drive\Yacht\Opdrachten\Hogeschool Utrecht\Repos\sean\user-app\input\{}.csv'.format(csvfile)
```
Comment (Fraukje) [inline user-app/scripts/readfiles.py]: Kun je deze file in de RD zetten? Ik push zo een commit voor deze functie zodat de pathname dynamisch is voor de user, file en projectnaam.

File: user-app/main.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+if __name__ == "__main__":
+    # load data from csv
+    from scripts.readfiles import readCsv
+    df = readCsv('tripadivsordata',nrows=100)
+
+    # run sentiment analysis 
+    from scripts.sentiment import sentAnalysisApp
+    sent_app = sentAnalysisApp(df,'review')
```
Comment (Fraukje) [inline user-app/main.py]: Calling the explicit column name is not very handy: maybe by default it will always use the second (or first column) in a csv file?

Comment (ALeemans) [reply user-app/main.py]: nee je moet dit ook niet zo zien, de main file is even tijdelijk todat we daadwerkelijk een interface hebben

Comment (Fraukje) [reply user-app/main.py]: Ah oke, ik denk dat ik dat nog niet goed begrijp of voor me kan zien :)

File: user-app/scripts/sentiment.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        colname = self.colname
+        labels = self.labels
+
+        # set model
+        model = AutoModelForSequenceClassification.from_pretrained(model_name)
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+  
+        # labels toevoegen als kolommen
+        for l in labels:
+            df[l] = 0.00
```
Comment (Fraukje) [inline user-app/scripts/sentiment.py:66]: This row results in len(df[l]) = 4 while len(labels) = 3. I don't understand why..? Does it make sense

File: user-app/scripts/sentiment.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+        for i in df.index:
+            # skip all text that have less then 30 or more than 900 characters
+            if len(df[colname][i]) > 15 and len(df[colname][i]) < 900:
```
Comment (Fraukje) [inline user-app/scripts/sentiment.py:70]: Can you add a justification for excluding <900 characters?

File: user-app/scripts/sentiment.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                output = model(**encoded_text)
+
+                # get scores from sentiment
+                scores = output[0][0].detach().numpy()
+                scores = softmax(scores)
+                
+                # if model uses a 5 points scale, recode to 3 points
+                if likert == 5:
+                    import numpy as np
+                    scores = np.array([scores[0:2].sum()]+[scores[2]]+[scores[3:5].sum()])
```
Comment (Fraukje) [inline user-app/scripts/sentiment.py:87]: Not sure if I understand correctly, but when likert == 5 you will have 5 scores right? so scores will contain values [0:4]

File: results/resultaten-versie-0.1.md [hunk 6]
```diff
@@ -74,25 +74,50 @@ Nu de mails geïdentificeerd en geschoond zijn kunnen we de sentiment analyse er
...
+## Betrouwbaarheidstests
+Alle modellen hebben we zelf getest door te kijken naar hotel reviews van een aantal hotels in Maastricht die we met een webscraper van [Tripadvisor.nl]('https://www.tripadvisor.nl/) gescraped hebben. We hebben in totaal naar 2.296 reviews van twee hotels gekeken. Deze reviews bevatten een korte review tekst en een review score van 1 tot 5. 
+We hebben vervolgens de review teksten door de drie verschillende modellen gehaald, daarnaast hebben we het 2e model (multilingual) omgezet naar een drie puntsschaal. Daarbij hebben we de steeds uiterste 2 waardes samen gevoegd (zeer positief wordt positief en idem voor negatief). Tot slot hebben we dit omgezette model samengoegd met het derde model (gemiddelde scores van de beide modellen samengevoegd). Hier kwamen de volgende resultaten uit voort:
+```ruby
+Accuracy test for sentiment analysis models on Tripadvisor data (n=2296)
```
Comment (Fraukje) [inline results/resultaten-versie-0.1.md:90]: Zou je deze info in een 3-by-3 tabel kunnen weergeven? Dat leest waarschijnlijk super chill af!

**Review by Fraukje — APPROVED** [review]
Very nice the sentAnalysisApp! Impressed with how smooth the whole script looks. I have some minor comments and nice-to-have improvements that are optional.
